http-mclient/test/Test/Network/HTTP/Mclient.hs

```diff
@@ -0,0 +1,219 @@
+module Test.Network.HTTP.Mclient (testTree) where
+
+import Control.Monad (when)
+import Control.Monad.Reader (asks)
+import Data.Word (Word8)
+import GHC.Enum (succ)
+import GHC.Records (HasField(..))
+import MIO.Core
+import MPrelude
+import Test.Tasty
+import Test.Tasty.HUnit ((@?=))
+
+import qualified Data.Aeson               as JSON
+import qualified Data.List                as List
+import qualified MIO.Log                  as Log
+import qualified Network.HTTP.Client      as HTTP
+import qualified Network.HTTP.MClient     as HTTP
+import qualified Network.HTTP.Types       as HTTP
+import qualified Network.Socket           as Network
+import qualified Network.Wai              as WAI
+import qualified Network.Wai.Handler.Warp as Warp
+import qualified Test.Tasty.HUnit         as Tasty
+import qualified UnliftIO
+import qualified UnliftIO.Concurrent      as UnliftIO
+
+data ExpectedResponse
+  = ConnectionFailure
+  | ResponseTimeout
+  | Status500
+  | Success
+  deriving stock (Eq, Show)
+
+data Environment = Environment
+  { httpSendRequest :: HTTP.SendRequest
+  , logAction       :: Log.Action
+  , requestsCounter :: UnliftIO.MVar Word8
+  }
+
+testTree :: TestTree
+testTree = testGroup "Network.HTTP.MClient"
+  [ testGroup "with a connection failure"
+      [ Tasty.testCase "return an HTTP connection failure after a single retry" .
+          runWithEnvironment $ do
+            (port, socket)  <- liftIO Warp.openFreePort
+
+            -- assert zero requests
```

Comment:
__suggestion: (nitpick)__ I think this comment is probably redundant given the function name.

Reply:
I agree. I will remove it.